integreat_cms/cms/models/abstract_content_translation.py

```diff
@@ -200,12 +200,23 @@ def available_languages(self):
         :rtype: dict
         """
         available_languages = {}
-        for language in self.foreign_object.public_languages:
-            if language == self.language:
-                continue
+        if self.foreign_object.region.use_fallback_language:
+            all_languages = self.foreign_object.region.active_languages
+        else:
+            all_languages = self.foreign_object.public_languages
+        for language in all_languages:
+            if not self.foreign_object.region.use_fallback_language:
+                if language == self.language:
+                    continue
             other_translation = self.foreign_object.get_public_translation(
                 language.slug
             )
+            if not other_translation and self.foreign_object.region.use_fallback_language:
+                other_translation = self.foreign_object.get_public_translation(
+                    self.foreign_object.region.default_language.slug
+                )
+                if other_translation:
+                    other_translation.language = self.foreign_object.region.default_language
```

Comment:
Keep in mind that the abstract content translation is used for all content types, but fallback translation should only be used for events and locations.